src/epplibwrapper/utility/pool.py

```diff
@@ -0,0 +1,133 @@
+import logging
+from typing import List
+import gevent
+from geventconnpool import ConnectionPool
+from epplibwrapper.socket import Socket
+from epplibwrapper.utility.pool_error import PoolError, PoolErrorCodes
+
+try:
+    from epplib.commands import Hello
+    from epplib.exceptions import TransportError
+except ImportError:
+    pass
+
+from gevent.lock import BoundedSemaphore
+from collections import deque
+
+logger = logging.getLogger(__name__)
+
+
+class EPPConnectionPool(ConnectionPool):
+    """A connection pool for EPPLib.
+
+    Args:
+        client (Client): The client
+        login (commands.Login): Login creds
+        options (dict): Options for the ConnectionPool
+        base class
+    """
+
+    def __init__(self, client, login, options: dict):
+        # For storing shared credentials
+        self._client = client
+        self._login = login
+
+        # Keep track of each greenlet
+        self.greenlets: List[gevent.Greenlet] = []
+
+        # Define optional pool settings.
+        # Kept in a dict so that the parent class,
+        # client.py, can maintain seperation/expandability
+        self.size = 1
+        if "size" in options:
+            self.size = options["size"]
+
+        self.exc_classes = tuple((TransportError,))
+        if "exc_classes" in options:
+            self.exc_classes = options["exc_classes"]
+
+        self.keepalive = None
+        if "keepalive" in options:
+            self.keepalive = options["keepalive"]
+
+        # Determines the period in which new
+        # gevent threads are spun up
+        self.spawn_frequency = 0.1
```

Comment:
**comment**: maybe some context here as to what the measurement is. is that .1 of a second?